src/lib/cli.ts

```diff
@@ -0,0 +1,103 @@
+import { app } from 'electron';
+import { executeWPCli } from '../../vendor/wp-now/src/execute-wp-cli';
+
+type CommandAction = 'wp';
+interface Command {
+	action: CommandAction;
+	args: string[];
+	executed: boolean;
+}
+
+const STANDALONE_COMMANDS: [ CommandAction ] = [ 'wp' ];
+const systemLog = console.log;
+let cliCommand: Command;
+
+export const isCLI = () => app.commandLine.hasSwitch( 'cli' );
+
+export const getCLIDataForMainInstance = () => {
+	if ( ! isCLI() ) {
+		return {};
+	}
+
+	return { cliCommand: app.commandLine.getSwitchValue( 'cli' ) };
+};
+
+export const listenCLICommands = () => {
+	app.on( 'second-instance', async ( event, commandLine, workingDirectory, additionalData ) => {
+		const data = additionalData as { cliCommand: string };
+		setCommand( data.cliCommand );
+		await executeCLICommand();
+	} );
+};
+
+export const processCLICommand = async ( {
+	mainInstance,
+	appBoot,
+}: {
+	mainInstance: boolean;
+	appBoot?: () => Promise< void >;
+} ) => {
+	setCommand( app.commandLine.getSwitchValue( 'cli' ) );
+
+	if ( STANDALONE_COMMANDS.includes( cliCommand.action ) ) {
+		await executeCLICommand();
+		app.quit();
+	} else if ( mainInstance && appBoot ) {
+		await appBoot();
+	} else {
+		app.quit();
+	}
+};
+
+export const executeCLICommand = async () => {
+	if ( cliCommand.executed ) {
+		return false;
+	}
+	cliCommand.executed = true;
+
+	await commands[ cliCommand.action ]?.();
+	return true;
+};
+
+const commands = {
+	wp: async () => {
+		disableConsole();
+		const { args } = cliCommand;
+		const projectPath =
+			args.find( ( arg ) => arg.startsWith( '--path=' ) )?.split( '=' )[ 1 ] || process.cwd();
+		const argsSansPath = args.filter( ( arg ) => ! arg.startsWith( '--path=' ) );
+
+		try {
+			const result = await executeWPCli( projectPath, argsSansPath );
+			if ( result.stderr ) {
+				systemLog( result.stderr );
+			}
+			systemLog( result.stdout );
```

Comment:
Also, colorization is currently not parsed or rendered on Windows.  

<img width="716" alt="windows-no-colorization" src="https://github.com/Automattic/studio/assets/438664/d70feaa9-469f-4c67-809f-45f553f7220c">

Reply:
> Also, colorization is currently not parsed or rendered on Windows.

The results might differ depending on the terminal used. When using Powershell, I noticed the colorization works but not with the standard terminal.

Reply:
As discovered in a pairing session, this problem comes from the fact that PHP is not able to detect the platform is running from. This implies that platform-specific, [e.g. Windows](https://github.com/wp-cli/wp-cli/blob/f701f406aa39f6aeca2af16e92712780e2675bad/php/utils.php#L699-L707), is not being executed and produces unexpected outcomes.

This will be solved in https://github.com/Automattic/studio/pull/213/commits/23433c4df259b68f2d53762970cabc3419dd758f, although it would be interesting to provide a better way to ensure we can specify the OS in WP-CLI.